src/components/NewDesign/Swap/Swap.tsx

```diff
@@ -0,0 +1,379 @@
+          .mul(pools[pairIndex].fee.val).div(new BN(10 ** pools[pairIndex].fee.scale)))
+        setPoolIndex(pairIndex)
+        console.log(printBN(new BN(10 ** pools[pairIndex].fee.scale).sub(new BN(1)), pools[pairIndex].fee.scale))
+      }
+    }
+  }, [tokenToIndex, tokenFromIndex, swap])
+
+  useEffect(() => {
+    swap ? setTokenToIndex(tokenToIndex) : setTokenToIndex(null)
+  }, [tokenFromIndex])
+
+  useEffect(() => {
+    swap ? setTokenFromIndex(null) : setTokenFromIndex(tokenFromIndex)
+  }, [tokenToIndex])
+
+  const getSwapPoolIndex = (fromToken: PublicKey, toToken: PublicKey) => {
+    return pools.findIndex((pool) => {
+      return (pool.tokenX.toString() === fromToken.toString() &&
+      pool.tokenY.toString() === toToken.toString()) ||
+      (pool.tokenX.toString() === toToken.toString() &&
+      pool.tokenY.toString() === fromToken.toString())
+    })
+  }
+  const getIsXToY = (fromToken: PublicKey, toToken: PublicKey) => {
+    const swapPool = pools.find(
+      pool =>
+        (fromToken.toString() === pool.tokenX.toString() &&
+          toToken.toString() === pool.tokenY.toString()) ||
+        (fromToken.toString() === pool.tokenY.toString() &&
+          toToken.toString() === pool.tokenX.toString())
+    )
+
+    return !!swapPool
+  }
+  const updateEstimatedAmount = (amount: string | null = null) => {
+    if (tokenFromIndex !== null && tokenToIndex !== null) {
+      setAmountTo(
+        calculateSwapOutAmount(tokens[tokenFromIndex], tokens[tokenToIndex], amount ?? amountTo)
+      )
+    }
+  }
+  const updateFromEstimatedAmount = (amount: string | null = null) => {
+    if (tokenFromIndex !== null && tokenToIndex !== null) {
+      setAmountFrom(
+        calculateSwapOutAmount(tokens[tokenToIndex], tokens[tokenFromIndex], amount ?? amountFrom)
+      )
+    }
+  }
+
+  const getButtonMessage = () => {
+    if (walletStatus !== Status.Initialized) {
+      return 'Please connect wallet'
+    }
+
+    if (tokenFromIndex === null || tokenToIndex === null) {
+      return 'Swap tokens'
+    }
+
+    if (!getIsXToY(tokens[tokenFromIndex].assetAddress, tokens[tokenToIndex].assetAddress)) {
+      return 'Pair does not exist'
+    }
+
+    if (
+      printBNtoBN(amountFrom, tokens[tokenFromIndex].decimal).eqn(0) ||
+      printBNtoBN(amountTo, tokens[tokenToIndex].decimal).eqn(0)
+    ) {
+      return 'Insufficient trade volume'
+    }
+    if (swap) {
+      if (
+        printBNtoBN(amountTo, tokens[tokenToIndex].decimal).gt(
+          tokens[tokenToIndex].balance
+        )
+      ) {
+        return 'Insufficient balance'
+      }
+    } else {
+      if (
+        printBNtoBN(amountFrom, tokens[tokenFromIndex].decimal).gt(
+          tokens[tokenFromIndex].balance
+        )
+      ) {
+        return 'Insufficient balance'
+      }
+    }
+    return 'Swap'
+  }
+
+  const setSlippage = (slippage: string): void => {
+    setSlippTolerance(slippage)
+    console.log(slippTolerance)
```

Comment:
remove this

Reply:
this is required for set slippage tolerance

Reply:
I've meant console.log

Reply:
done